Work in progress: 0.18 release notes

bep opened the pull request:
@moorereason @digitalcraftsman @spf13 @anthonyfok and others ... this is just a rough outline. Please edit / add.

digitalcraftsman commented:
Isn't it worth noting that nodes have been replaced by pages? As far as I know the changes you made are backwards compatible. But certain things might be handled differently by theme maintainers if they are aware of this change.

I glimpsed over the last commits and didn't find this pull requests on the list:

- new sha256 template function (#2762)
- new instagram shortcode (#2690)

bep commented:
>Isn't it worth noting that nodes have been replaced by pages? 

Yes that is the main feature -- but I added that to the release notes when I added the feature (fresh memory), so they do not show up on this diff.

I browsed through the commit log, but I obviously missed something. I have checked the "allow edits from other maintainers" -- couldn't you just update the file (with the new items you found + formatting + editing, if you have time ...)?

We have a week++ until release so no rush.

digitalcraftsman commented:
Sure, I'll add them. I'm not in hurry but just wanted to highlight the additions rather sooner than later :wink:

bep commented:
I notice that some of the "fixes" is really "improvements", but we will fix that, eventually.

bep commented on docs/content/meta/release-notes.md:
```diff
@@ -42,25 +46,32 @@ Not only is this a much simpler model to understand, it is also faster and paved
 ## Other New Features
 
+* Every `Page` now has a `Kind` property. Since everything is a `Page` now, the `Kind` is used to differentiate different kinds of pages.
+  Possible values are `page`, `home`, `section`, `taxonomy`, `taxonomyTerm`, `RSS`, `sitemap`, `robotsTXT`, and `404`.
```
This is true, but it may confuse some users. I deliberately kept all from RSS and right unexported, because they are only of internal interest (for now) -- these pages does not show up in any of the page collections and as such these kinds cannot be used for anything practical for end users. So I suggest just mentioning those in parens or something)

moorereason replied:
I'll tweak it shortly.  Thanks

bep commented:
We should add some lines about the performance:

```
benchmark           old ns/op       new ns/op       delta
BenchmarkHugo-4     30747531411     10372935501     -66.26%

benchmark           old allocs     new allocs     delta
BenchmarkHugo-4     88370619       43654211       -50.60%

benchmark           old bytes       new bytes      delta
BenchmarkHugo-4     12431663920     9200452080     -25.99%
```

I have run the numbers twice, and the error margin seems to be just a couple of percent on my MacBook, so this is impressive.

This is https://github.com/bep/hugo-benchmark 

Maybe add one line in the same breath about  the `partialCached`, which can be very good for complex but static partials.
